libbeat/common/schema/mapstriface/mapstriface.go

```diff
@@ -193,9 +193,43 @@ func toInteger(key string, data map[string]interface{}) (interface{}, error) {
 		if err == nil {
 			return int64(f64), nil
 		}
-		return 0, fmt.Errorf("Expected integer, found json.Number (%v) that cannot be converted", num)
+		return 0, fmt.Errorf("expected integer, found json.Number (%v) that cannot be converted", num)
 	default:
-		return 0, fmt.Errorf("Expected integer, found %T", emptyIface)
+		return 0, fmt.Errorf("expected integer, found %T", emptyIface)
+	}
+}
+
+// Float creates a Conv object for converting floats. Acceptable input
+// types are int64, int, and float64.
+func Float(key string, opts ...schema.SchemaOption) schema.Conv {
+	return schema.SetOptions(schema.Conv{Key: key, Func: toFloat}, opts)
+}
+
+func toFloat(key string, data map[string]interface{}) (interface{}, error) {
+	emptyIface, exists := data[key]
+	if !exists {
+		return 0, fmt.Errorf("Key %s not found", key)
+	}
+	switch emptyIface.(type) {
+	case float64:
```

Comment:
Should `string` be accepted?

Reply:
+1 on parsing string via `strconv.ParseFloat`.

Reply:
So far we don't need it and it's also not supported in the Int type. I would add it when we need it.

Reply:
That’s fine by me to omit it for now. It’s an easy add at any point.